code/datums/ai/kitchenbot/kitchenbot_behaviors.dm

```diff
@@ -0,0 +1,138 @@
+
+//Unless specifically mentioned, these are usable on other mobs (but will ofc be themed as kitchenbot)
+
+//CLEANING
+
+/datum/ai_behavior/find_and_set/find_disposals
+	locate_path = /obj/machinery/disposal/bin
+	bb_key_to_set = BB_KITCHENBOT_CHOSEN_DISPOSALS
+
+/datum/ai_behavior/find_and_set/find_disposals/react_to_success(datum/ai_controller/controller)
+	controller.pawn.audible_message("<span class='hear'>[controller.pawn] makes a chiming sound! It must have located a disposal bin.</span>")
+	playsound(controller.pawn, 'sound/machines/chime.ogg', 50, FALSE)
+
+//kitchenbot because it uses some fail keys, could def be made normal
+/datum/ai_behavior/find_and_set/find_refuse
+	bb_key_to_set = BB_KITCHENBOT_TARGET_TO_DISPOSE
+
+/datum/ai_behavior/find_and_set/find_refuse/search_tactic(datum/ai_controller/controller)
+	var/obj/item/found_refuse
+	for(var/obj/object in oview(7, controller.pawn))
+		if(istype(object, /obj/item/trash/plate)) //dirty dishes
+			found_refuse = object
+		else if(istype(object, /obj/item/reagent_containers/food/condiment)) //bags of sugar flour etc, dump if empty
+			var/obj/item/reagent_containers/food/condiment/condiment_container = object
+			if(!condiment_container.reagents || !condiment_container.reagents.total_volume) //EMPTY! TRASH! GARBAGE! REFUUUUUSE
+				found_refuse = condiment_container
+	return found_refuse
+
+/datum/ai_behavior/find_and_set/find_refuse/react_to_success(datum/ai_controller/controller)
+	controller.blackboard[BB_KITCHENBOT_FAILED_LAST_TARGET_SEARCH] = FALSE //we found some trash, go start using the fast version
+
+/datum/ai_behavior/find_and_set/find_refuse/fast
```

Comment:
clever, good job.

Reply:
removed it for an even cleverer idea (don't find individual plates, just search for new ones and maintain a list)